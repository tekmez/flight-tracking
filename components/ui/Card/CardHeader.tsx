import { View, Text } from "react-native";
import React from "react";
import Badge from "../Badge";
import Airline from "../AirlineIcon";

type CardHeaderProps = {
  airline: string;
  duration: string;
  flightNumber: string;
};

const CardHeader = ({ airline, duration, flightNumber }: CardHeaderProps) => {
  const iconName = airline.split(" ")[0] as
    | "Turkish"
    | "Ryanair"
    | "Emirates"
    | "Lufthansa";
  return (
    <View className="flex-row justify-between items-center mb-4">
      <View className="flex-row items-center">
        <Airline name={iconName} />
        <Text className="font-sfProMedium text-[13px] leading-[18px] text-center text-gray-500 my-[3px] ml-2">
          {flightNumber} â€¢ {airline}
        </Text>
      </View>
      <Badge duration={duration} />
    </View>
  );
};

export default CardHeader;
