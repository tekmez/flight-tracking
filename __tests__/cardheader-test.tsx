import React from "react";
import { render } from "@testing-library/react-native";
import CardHeader from "@/components/ui/Card/CardHeader";

describe("CardHeader", () => {
  const defaultProps = {
    airline: "Turkish Airlines",
    duration: "2h 30m",
    flightNumber: "TK1234",
  };

  it("renders correctly", () => {
    const { toJSON } = render(<CardHeader {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("displays correct airline and flight number", () => {
    const { getByText } = render(<CardHeader {...defaultProps} />);
    expect(getByText("TK1234 â€¢ Turkish Airlines")).toBeTruthy();
  });

  it("renders Badge component with correct duration", () => {
    const { getByText } = render(<CardHeader {...defaultProps} />);
    expect(getByText("2h 30m")).toBeTruthy();
  });

  it("renders Airline component", () => {
    const { getByTestId } = render(<CardHeader {...defaultProps} />);
    expect(getByTestId("airline-icon")).toBeTruthy();
  });
});
