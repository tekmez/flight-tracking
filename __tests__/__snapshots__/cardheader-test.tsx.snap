// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
      {
        "alignItems": "center",
      },
      {
        "marginBottom": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <RNSVGSvgView
      bbHeight={24}
      bbWidth={24}
      fill="none"
      focusable={false}
      height={24}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="airline-icon"
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M13.25.08C19.28.713 24 5.791 24 12.02A11.97 11.97 0 0 1 12.02 24C5.395 24 0 18.645 0 12.02S5.355 0 12.02 0c.396 0 .833.04 1.23.08Zm.634 23.087c6.15-1.032 10.314-6.863 9.283-13.051A11.314 11.314 0 0 0 13.21.754c-.992-.12-2.023-.08-3.055.08a10.83 10.83 0 0 0-3.411 1.15c5.395 1.864 8.33 4.68 8.529 7.537.119 1.785-.714 3.054-1.825 4.086l7.696-.516c.198 0 .278.238.04.317L2.816 18.645a11.225 11.225 0 0 0 7.973 4.601c1.032.12 2.063.12 3.094-.08Zm-5.038-10.91C9.838 8.49 8.688 4.523 5.95 2.42 2.221 4.8.04 9.243.833 13.884A10.595 10.595 0 0 0 2.42 18.05c2.935-.794 5.514-2.34 6.426-5.792Z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M13.884 23.167c6.15-1.032 10.314-6.863 9.283-13.051A11.314 11.314 0 0 0 13.21.754c-.992-.12-2.023-.08-3.055.08a10.83 10.83 0 0 0-3.411 1.15c5.395 1.864 8.33 4.68 8.529 7.537.119 1.785-.714 3.054-1.825 4.086l7.696-.516c.198 0 .278.238.04.317L2.816 18.645a11.225 11.225 0 0 0 7.973 4.601c1.032.12 2.063.12 3.094-.08Zm-5.038-10.91C9.838 8.49 8.688 4.523 5.95 2.42 2.221 4.8.04 9.243.833 13.884A10.595 10.595 0 0 0 2.42 18.05c2.935-.794 5.514-2.34 6.426-5.792Z"
          fill={
            {
              "payload": 4291234316,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        [
          {
            "fontFamily": "SFProMedium",
          },
          {
            "fontSize": 13,
          },
          {
            "lineHeight": 18,
          },
          {
            "textAlign": "center",
          },
          {
            "color": "#6b7280",
          },
          {
            "marginBottom": 3,
            "marginTop": 3,
          },
          {
            "marginLeft": 8,
          },
        ]
      }
    >
      TK1234
       â€¢ 
      Turkish Airlines
    </Text>
  </View>
  <View
    style={
      [
        {
          "paddingBottom": 4,
          "paddingTop": 4,
        },
        {
          "paddingLeft": 10,
          "paddingRight": 10,
        },
        {
          "borderBottomWidth": 0.5,
          "borderLeftWidth": 0.5,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
        },
        {
          "borderBottomLeftRadius": 7,
          "borderBottomRightRadius": 7,
          "borderTopLeftRadius": 7,
          "borderTopRightRadius": 7,
        },
        {
          "borderBottomColor": "#fed7aa",
          "borderLeftColor": "#fed7aa",
          "borderRightColor": "#fed7aa",
          "borderTopColor": "#fed7aa",
        },
        {
          "backgroundColor": "#fff7ed",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "SFProMedium",
          },
          {
            "fontSize": 12,
          },
          {
            "lineHeight": 16,
          },
          {
            "color": "#c2410c",
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      2h 30m
    </Text>
  </View>
</View>
`;
